-1 
Tested at: 952fdcca3238fe2a2f489c61106df08284acaf4e
 I found a compilation error while trying to compile with clang: 
 
 <pre>                               ^
/afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc491/external/boost/1.51.0-cms/include/boost/preprocessor/cat.hpp:29:34: note: expanded from macro 'BOOST_PP_CAT_I'
#    define BOOST_PP_CAT_I(a, b) a ## b
                                 ^
<scratch space>:159:1: note: In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-02-28-1400/src/JetMETCorrections/Type1MET/plugins/CorrectedCaloMETProducer.cc:21:
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-02-28-1400/src/JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h:98:76: error: default initialization of an object of const type 'const CorrectedMETProducer_namespace::CorrectedMETFactoryT<CaloMET>' without a user-provided default constructor
      static const CorrectedMETProducer_namespace::CorrectedMETFactoryT<T> correctedMET_factory;
                                                                           ^
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-02-28-1400/src/JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h:70:12: note: in instantiation of member function 'CorrectedMETProducerT<reco::CaloMET>::produce' requested here
  explicit CorrectedMETProducerT(const edm::ParameterSet& cfg)
           ^
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8004/2998/summary.html

--- a/JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h
+++ b/JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h
@@ -52,6 +52,7 @@ namespace CorrectedMETProducer_namespace
   class CorrectedMETFactoryT
   {
     public:
+     explicit CorrectedMETFactoryT() {}
 
      T operator()(const T&, const CorrMETData&) const
      {

--- a/JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h
+++ b/JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h
@@ -52,7 +52,6 @@ namespace CorrectedMETProducer_namespace
   class CorrectedMETFactoryT
   {
     public:
-     explicit CorrectedMETFactoryT() {}
 
      T operator()(const T&, const CorrMETData&) const
      {
@@ -96,7 +95,7 @@ class CorrectedMETProducerT : public edm::EDProducer
 	  rawMEt != rawMEtCollection->end(); ++rawMEt ) {
       CorrMETData correction = algorithm_->compMETCorrection(evt, es);
       
-      static const CorrectedMETProducer_namespace::CorrectedMETFactoryT<T> correctedMET_factory;
+      static const CorrectedMETProducer_namespace::CorrectedMETFactoryT<T> correctedMET_factory {};
       T correctedMEt = correctedMET_factory(*rawMEt, correction);
 
       correctedMEtCollection->push_back(correctedMEt);

--- a/JetMETCorrections/Type1MET/interface/PFJetMETcorrInputProducerT.h
+++ b/JetMETCorrections/Type1MET/interface/PFJetMETcorrInputProducerT.h
@@ -151,13 +151,13 @@ class PFJetMETcorrInputProducerT : public edm::EDProducer
     for ( int jetIndex = 0; jetIndex < numJets; ++jetIndex ) {
       const T& rawJet = jets->at(jetIndex);
 
-      const static PFJetMETcorrInputProducer_namespace::InputTypeCheckerT<T, Textractor> checkInputType;
+      const static PFJetMETcorrInputProducer_namespace::InputTypeCheckerT<T, Textractor> checkInputType {};
       checkInputType(rawJet);
 
       double emEnergyFraction = rawJet.chargedEmEnergyFraction() + rawJet.neutralEmEnergyFraction();
       if ( skipEM_ && emEnergyFraction > skipEMfractionThreshold_ ) continue;
 
-      const static PFJetMETcorrInputProducer_namespace::RawJetExtractorT<T> rawJetExtractor;
+      const static PFJetMETcorrInputProducer_namespace::RawJetExtractorT<T> rawJetExtractor {};
       reco::Candidate::LorentzVector rawJetP4 = rawJetExtractor(rawJet);
       if ( skipMuons_ ) {
 	const std::vector<reco::CandidatePtr> & cands = rawJet.daughterPtrVector();

